### DISCUSSION TITLE
Protocol parameter in profile schema to be added in CLI

### DISCUSSION THREAD

**Discussion by zdmullen** [zowe/zowe-explorer-vscode#1189] [Ideas]
FYI, there is an upcoming enhancement to the CLI where the --protocol is being exposed to all commands and also made available in the profile schema. That said, I would like to get your thoughts on a few ways we can adopt that enhancement.

Maybe this 2-part question will help shape the conversation:
Do we need to ask for a fully qualified host to start with?
<protocol>://<host>:<port>
what about <protocol>://<user>:<password>@<host>:<port> ?

**Comment by zdmullen** [comment]
Responses from Billie:
If it will be included in the schema we will see it upon creation/editing of profiles, and if protocol isn't added to the host textbox but is a field in the schema another textbox will pop up asking for it.
This is the same with the port value.
We may need to adjust the code to where if it is included in the host textbox to save that value and possibly not prompt for it a second time.
Plus it is already stated in the <protocol>://<host>:<port>  manner so we would really just need to capture the input value and use it, since I don't believe that is happening at this time with previous way of handling zosmf profiles.

**Comment by jellypuno** [comment]
Note: A good candidate for next PI and/or the ZE Beta Version
